sdk/python/foundation-models/system/inference/image-text-to-text/image-text-to-text-online-endpoint.ipynb

```diff
@@ -0,0 +1,440 @@
+    "\n",
+    "    conv = conv_templates[conv_mode].copy()\n",
+    "    roles = conv.roles\n",
+    "\n",
+    "    first_time = True\n",
+    "    while True:\n",
+    "        try:\n",
+    "            inp = input(f\"{roles[0]}: \")\n",
+    "        except EOFError:\n",
+    "            inp = \"\"\n",
+    "        if not inp:\n",
+    "            print(\"exit...\")\n",
+    "            break\n",
+    "\n",
+    "        print(f\"{roles[1]}: \", end=\"\")\n",
+    "\n",
+    "        if first_time:\n",
+    "            first_time = False\n",
+    "            inp = DEFAULT_IMAGE_TOKEN + \"\\n\" + inp\n",
+    "            conv.append_message(conv.roles[0], inp)\n",
+    "        else:\n",
+    "            # later messages\n",
+    "            conv.append_message(conv.roles[0], inp)\n",
+    "        conv.append_message(conv.roles[1], None)\n",
+    "        prompt = conv.get_prompt()\n",
+    "\n",
+    "        make_request_prompt(image_bytes, prompt)\n",
+    "        response = get_response()\n",
+    "\n",
+    "        conv.messages[-1][-1] = response\n",
+    "\n",
+    "        printed_response = response\n",
+    "        if printed_response.endswith(\"<|im_end|>\"):\n",
+    "            printed_response = printed_response[: -len(\"<|im_end|>\")]\n",
+    "        if printed_response.endswith(\"</s>\"):\n",
+    "            printed_response = printed_response[: -len(\"</s>\")]\n",
+    "        print(printed_response)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### 5.1. Direct question\n",
+    "\n",
+    "We provide functionality to directly submit a question to the model about an image, internally converting the question to the prompt format required by the model version. This is done by setting the \"prompt\" column of the input to the empty string and the \"direct_question\" column to the question, see the `make_request_direct_question()` function above.\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "image_bytes = open(os.path.join(dataset_dir, \"milk_bottle\", \"99.jpg\"), \"rb\").read()\n",
+    "question = \"What is in this image?\"\n",
+    "\n",
+    "\n",
+    "make_request_direct_question(image_bytes, question)\n",
+    "print(get_response())"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### 5.2. Dialog\n",
+    "\n",
+    "We provide functionality to submit a prompt to the model, which can encode either a question or a dialog ending with a question for the model. This is done by setting the \"prompt\" column of the input, see the `make_request_prompt()` function above; note that the \"direct_question\" column is ignored in this case. Below, we rely on functionality in the `llava-torch` package to manage the dialog and convert it into a prompt for the model, see the `run_dialog_loop()` function above.\n"
```

Comment:
Nit: I think it would be more readable to rearrange the the cells so that the make_request_direct_question call is grouped with the related functions and likewise for the make_request_prompt

Reply:
Addressed.